### PR TITLE
[RFC][Core] Displaying payment error message when payment is failed.

### PR DESCRIPTION
**Pull request by ahmadrabie**
Q  | A
------------- | -------------
Bug fix?  | no
New feature?  | yes
BC breaks? | no
Deprecations? | no
Fixed tickets | -
License | MIT
Doc PR | -

fixes: #2678

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Sounds good although I am not sure it is responsibility of state machine to display flashes... We should think more about it.

**Comment by ahmadrabie** [comment]
I agree and I have the same feeling but unfortunately I couldn't find a better solution, maybe @makasim can help here!

File: src/Sylius/Bundle/CoreBundle/StateMachineCallback/FailPaymentCallback.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        $this->session = $session;
+        $this->translator = $translator;
+    }
+
+    public function addFailFlash(PaymentInterface $payment)
+    {
+
+        $details = $payment->getDetails();
+        if (isset($details['error']) && !empty($details['error'])) {
```
Comment (makasim) [inline src/Sylius/Bundle/CoreBundle/StateMachineCallback/FailPaymentCallback.php]: This is not correct. Payment details contains details in a payment gateway specific format, and hence they are different for paypal and stripe for example and may not have `error` key.

**Comment by ahmadrabie** [comment]
:+1:

**Pull request closed without merging**